azure-pipelines.yml

```diff
@@ -225,7 +225,7 @@ stages:
                 $env:ARM_SUBSCRIPTION_ID="$(azure_subscription_id)"
                 cd $(Pipeline.Workspace)\drop_terraform\environments\dev
                 terraform init
-                terraform plan -no-color -input=false -var="HMAC_SIGNING_KEY=$(HMAC_SIGNING_KEY)" -var="API_KEYS=$(API_KEYS)" -var="APIM_SUBSCRIPTION_KEY=$(APIM_SUBSCRIPTION_KEY)"
+                terraform plan -no-color -input=false -var="HMAC_SIGNING_KEY=$(HMAC_SIGNING_KEY)" -var="API_KEYS=$(API_KEYS)" -var="APIM_SUBSCRIPTION_KEY=$(APIM_SUBSCRIPTION_KEY)" -var="gov_notify_api_key=$(gov_notify_api_key)" -var="user_roles_changed_email_template_id=$(user_roles_changed_email_template_id)"
```

Comment:
Is user_roles_changed_email_template_id a secret? If it is just a template id the it can be passed in as a normal TF_VAR. This pattern is only used for secret variables due to the way azdo secret pipeline variables work.

Reply:
On the topic of template IDs, is it still the case that Gov.Notify offers us only a flat registry across all environments? Have we got a strategy for how we manage making updates to existing templates without affecting multiple environments?

Reply:
I believe the fact that we can specify the template id in the vars file, which can be environment specific should give us what we need

Reply:
correct, we can have multiple templates and can generate our own API keys with different permission sets